src/Compilers/Core/Portable/Diagnostic/DiagnosticInfo.cs

```diff
@@ -137,7 +137,7 @@ internal DiagnosticInfo(CommonMessageProvider messageProvider, bool isWarningAsE
         }
 
         // Create a copy of this instance with a explicit overridden severity
-        internal DiagnosticInfo GetInstanceWithSeverity(DiagnosticSeverity severity)
+        internal virtual DiagnosticInfo GetInstanceWithSeverity(DiagnosticSeverity severity)
```

Comment:
Half of me wants to make this method `abstract` to force all derived types to acknowledge this potential issue. The other half of me thinks that's a bit overkill here.

Reply:
Yeah, it makes me wish for the ability to say "all of these members in the group must be overridden, or none of them".